week-6/Homework/mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,7 +1,59 @@
 // Implement the missing functionality according to the task description.
 // Hint: Remember which function you need to call when it's time to alarm.
+/* 
+let counter = Number;
+let alarmSet = document.querySelector("#alarmSet");
+alarmSet.html('0');
+let timer = document.querySelector("#timeRemaining");
+
+document.querySelector("#set").addEventListener("click",()=>{
+  setInterval(setAlarm, 1000);
+})
+
+document.querySelector("#stop").addEventListener("click",()=>{
+  clearInterval(clear);
+})
+
+function setAlarm() { 
+  
+counter--;
+alarmSet.innerHTML  =  counter;
+timer.innerText = ("time remaining: 00.0" + counter);
+if (counter === 0){
+  playAlarm()
+}
+} */
+ 
+function setAlarm() {
```

Comment:
Good effort, but I think you overcomplicated it a bit. There is no need for a for loop, or for any additional event listeners.

Have a look at Mohammed's solution, which is quite simple but it does the job. You might find it easier to understand than the official solution:
https://github.com/CodeYourFuture/js-exercises/blob/31c8642d3646e002e62290db41196d787df74aee/week-6/Homework/mandatory/1-alarmclock/alarmclock.js